### PR TITLE
(DRAFT) mergeStyles loosening typings & top level selectors support

### PR DESCRIPTION
**Pull request by aneeshack4**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Loosening type safety from styles to support css variables & adding support for &: selectors as top level entities. This is not fully done yet - just opening a draft PR to start getting feedback & ideas.

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10253)

### CODE REVIEW

File: packages/merge-styles/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  // Use IntelliSense to learn about possible attributes.
+  // Hover to view descriptions of existing attributes.
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+  "version": "0.2.0",
```
Comment (aneeshack4) [inline packages/merge-styles/.vscode/launch.json]: Ignore this - will remove this & the other vscode settings json file in actual PR.I

File: packages/merge-styles/src/mergeStyleSets.test.ts [hunk 2]
```diff
@@ -95,6 +95,88 @@ describe('mergeStyleSets', () => {
...
+    // expect(_stylesheet.getRules()).toEqual('.a-0:hover{background:green;}');
+  });
+
+  it('can expand child selectors with static class names', () => {
+    const styles = mergeStyleSets({
+      root: [
+        'a',
+        {
+          selectors: {
+            '&:hover $child': { background: 'red' }
```
Comment (dzearing) [inline packages/merge-styles/src/mergeStyleSets.test.ts]: this test is a little odd, as we don't support the `$child` syntax any more.

File: packages/merge-styles/src/extractRules.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        } else {
+          // tslint:disable-next-line:no-any
+          (currentRules as any)[prop] = (arg as any)[prop] as any;
+        }
+      }
+    }
+  }
+}
+
+function expandQuads(currentRules: IDictionary, name: string, value: string): void {
```
Comment (dzearing) [inline packages/merge-styles/src/extractRules.ts]: perhaps each of these functions should live separately

**Comment by dzearing** [comment]
Could we finalize this? This was closed without reason and it took me by surprise that `selectors` is still required.

**Comment by aneeshack4** [comment]
@dzearing @jdhuntington and I paired on this and had to pause because other work came up. The mergestyles code is quite obscure and hard to understand just by reading it so we had to write a lot of tests and breakpoint debug and this only scraped the surface if I remember correctly. If we're to start this up again, we should pair and document it better this time/make it more readable.

**Pull request closed without merging**